Plans: update `plan-storage` component

retrofox opened the pull request:
In this PR we do a refactoring of the `<PlanStorage />` component. In the issue #11508 there are information about this.

Anyway, I'll paste the primary tasks:

 Move component into b

- locks/ place
-  rewrite component using class
-  It's no longer like a button.
-  No plan name.
-  It has an explicit link "Upgrade" that takes you to /plans
-  Clicking on other parts of the component, no longer take you to /plans
-  We don't show it at all for the Business plan because there is no upgrade and the storage is unlimited.

### Testing

You could test the component in its blocks testing page:
http://calypso.localhost:3000/devdocs/blocks/plan-storage

<img src="https://cloud.githubusercontent.com/assets/77539/23281138/dad99f7e-f9fa-11e6-8dff-5b8df424c674.png" width="300px" />

Pay attention that we've added a new PlanStorageBlock instance in the example page which hs defined the plan as `Business`. The component shouldn't render anything here.

Also, you can take a look how looks the component into the media modal:

1) Start to write/edit a post: `http://calypso.localhost:3000/post/<testing-site>`
2) Add media files.
<img src="https://cloud.githubusercontent.com/assets/77539/23280658/dcc0ffbe-f9f8-11e6-890a-8d1172a96646.png" width="300px" />

3) Take a look at the new presentation of the component.

<img src="https://cloud.githubusercontent.com/assets/77539/23281171/fafa9eac-f9fa-11e6-947e-f72f374e0fd4.png" width="300px" />

**spanish**
<img src="https://cloud.githubusercontent.com/assets/77539/23281187/10ca05a6-f9fb-11e6-9ae5-b91fda25ab9c.png" width="300px" />

No render component for site with a Business plan
<img src="https://cloud.githubusercontent.com/assets/77539/23280757/334cf6e4-f9f9-11e6-800a-72de846b483b.png" width="300px" />

a8ci18n commented on client/blocks/plan-storage/block.jsx:
```diff
@@ -0,0 +1,90 @@
+		if ( sitePlanName === 'Premium' ) {
+			return translate( 'Premium Plan', { context: 'Short plan name on storage block' } );
+		} else if ( sitePlanName === 'Free' ) {
+			return translate( 'Free Plan', { context: 'Short plan name on storage block' } );
```
Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Free Plan', { context: 'Short plan name on storage button'} )` *ES Score: 16*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-component)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/blocks/plan-storage/block.jsx:
```diff
@@ -0,0 +1,90 @@
+		const { sitePlanName, translate } = this.props;
+
+		if ( sitePlanName === 'Premium' ) {
+			return translate( 'Premium Plan', { context: 'Short plan name on storage block' } );
```
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Premium Plan', { context: 'Short plan name on storage button'} )` *ES Score: 18*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-component)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/blocks/plan-storage/block.jsx:
```diff
@@ -0,0 +1,91 @@
+			return translate( 'Free Plan', { context: 'Short plan name on storage block' } );
+		}
+		//This is a fallback if we add a new plan. We ideally want to add any plan levels for proper i18n.
+		return translate( '%(planName)s Plan', { args: { planName: sitePlanName } } );
```
Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Free Plan', { context: 'Short plan name on storage button'} )` *ES Score: 16*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-component)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/blocks/plan-storage/block.jsx:
```diff
@@ -0,0 +1,91 @@
+			return translate( 'Premium Plan', { context: 'Short plan name on storage block' } );
+		} else if ( sitePlanName === 'Free' ) {
+			return translate( 'Free Plan', { context: 'Short plan name on storage block' } );
+		}
```
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Premium Plan', { context: 'Short plan name on storage button'} )` *ES Score: 18*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-component)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

gwwar commented on client/blocks/plan-storage/block.jsx:
```diff
@@ -0,0 +1,86 @@
+const ALERT_PERCENT = 80;
+const WARN_PERCENT = 60;
+
+class PlanStorageBlock extends Component {
```
Since this is in the blocks folder, calling this `PlanStorageBlock` is a bit redundant.

I recommend keeping this file as index.jsx and instead naming this `PlanStorage`

retrofox replied:
it's the old `button` internal component.

retrofox replied:
`<PlanStorageButton />`

gwwar replied:
Ah got it, though I'm not sure what to call it. Let's avoid block since it's an overloaded term.

PlanStorageCard maybe @iamtakashi ?

iamtakashi replied:
I'm not familiar with naming convention, but by looking at the conversation, I guess we can't call it just `PlanStorage`. What about `PlanStorageBar`?

retrofox replied:
I like `PlanStorageBar`, since the card has another implication in our environment IMO.

retrofox replied:
updated and rebased with `PlanStorageBar`

gwwar commented on client/blocks/plan-storage/index.jsx:
```diff
@@ -2,51 +2,47 @@
-					sitePlanName={ this.props.site.plan.product_name_short }
+				<QueryMediaStorage siteId={ siteId } />
+				<PlanStorageBar
+					siteSlug={ site.Slug }
```
Typo here? Clicking on the upgrade link currently breaks the page.

retrofox replied:
😨

gwwar commented on client/blocks/plan-storage/index.jsx:
```diff
@@ -2,51 +2,47 @@
-		if ( ! this.props.site || this.props.site.jetpack ) {
+		const { site, siteId } = this.props;
+
+		if ( ! site || site.jetpack ) {
```
Let's use more specific selectors, and pass those values through in props. For example: `isJetpackSite`, `getSiteSlug`

gwwar commented on client/blocks/plan-storage/index.jsx:
```diff
@@ -2,51 +2,47 @@
+				<QueryMediaStorage siteId={ siteId } />
+				<PlanStorageBar
+					siteSlug={ site.Slug }
+					sitePlanName={ site.plan.product_name_short }
```
Similar thing here, we can use `getSitePlan` and probably update this to use plan constants instead in PlanStorageBar

gwwar commented:
From the mocks it looks like "Upgrade" should have no text-decoration
<img width="221" alt="screen shot 2017-02-24 at 2 10 18 pm" src="https://cloud.githubusercontent.com/assets/1270189/23322891/01bf0138-fa9b-11e6-929b-a3a9c77e0aa6.png">

gwwar commented:
Looks like http://calypso.localhost:3000/devdocs/blocks is broken too, but overall looks relatively close. How are we feeling about updating the plan storage styling ahead of the action bar changes? I personally think it looks ok in the bottom left and can probably land in master rather than out feature branch.

@iamtakashi when you have time, can you 👍 or 👎 for design?

retrofox commented:
> Looks like http://calypso.localhost:3000/devdocs/blocks is broken

Fixed

> How are we feeling about updating the plan storage styling ahead of the action bar changes? I personally think it looks ok in the bottom left and can probably land in master rather than out feature branch.

It actually looks pretty good beyond of the new action bar improvements that we are gonna to do. Personally, I think it could be merged now. The design is clean and fresh.

![image](https://cloud.githubusercontent.com/assets/77539/23323853/56290218-fac9-11e6-97c1-c90adfa0091b.png)

![image](https://cloud.githubusercontent.com/assets/77539/23323871/636237d8-fac9-11e6-9f09-53d5672a9f0a.png)

Already updated and rebased.

gwwar commented on client/blocks/plan-storage/docs/example.jsx:
```diff
@@ -8,8 +8,14 @@ import PureRenderMixin from 'react-pure-render/mixin';
+	PLAN_PREMIUM,
+	PLAN_PERSONAL,
+	PLAN_FREE,
+} from 'lib/plans/constants';
```
⭐️

gwwar commented on client/blocks/plan-storage/index.jsx:
```diff
@@ -0,0 +1,65 @@
+	static propTypes = {
+		className: PropTypes.string,
+		mediaStorage: PropTypes.object,
+		siteId: PropTypes.number.isRequired,
```
On an empty cache the console might throw an error. How about we relax this, and instead render null when siteId is not available.

gwwar commented on client/blocks/plan-storage/style.scss:
```diff
@@ -0,0 +1,38 @@
+}
+
+.plan-storage__storage-link {
+	float: right;
```
We shouldn't need this since a span already has a default display of `inline-block`

gwwar replied:
Speaking of which I'd recommend moving this class one layer down to the actual link instead of the span wrapper, so we can avoid a bare element rule

gwwar commented on client/blocks/plan-storage/docs/example.jsx:
```diff
@@ -46,20 +48,34 @@ export default React.createClass( {
+
+				<div style={ { marginBottom: 16 } }>
+					<span style={ { fontSize: 12, color: 'grey' } }>
+						No render PlanStorage here since it's a business plan.
```
How about `Business plans have unlimited storage, so PlanStorage will not be rendered.`

iamtakashi commented:
LGTM in design wise 👍

retrofox commented:
According to the comments related with the link tag, etc. I defined the width of the component auto-expandable so. It means that the width will be adjusted depending on the size of wrapper element.

And since the `Upgrade` element is floating at the right, the whole visualization looks nice. 

<img src="https://cloud.githubusercontent.com/assets/77539/23368050/3dfe2122-fceb-11e6-8930-dce58ed97c8b.gif" width="500px" />


**Media modal library**
<img src="https://cloud.githubusercontent.com/assets/77539/23367940/e691ff08-fcea-11e6-8e88-a322fdd13ec8.png" width="300px" />

retrofox commented:
Thanks Kerry and Takashi for the review.

gwwar commented:
> And since the Upgrade element is floating at the right,

;) Maybe look into using flexbox next time @retrofox. I'd like us to avoid floats if possible since they're always a pain to clear.
